qiskit/pulse/ir/ir.py

```diff
@@ -66,10 +66,7 @@ def inst_targets(self) -> set[PulseTarget | Frame | MixedFrame]:
         """Recursively return a set of all Instruction.inst_target in the SequenceIR"""
         inst_targets = set()
         for elm in self.elements():
-            if isinstance(elm, SequenceIR):
-                inst_targets |= elm.inst_targets
-            else:
-                inst_targets.add(elm.inst_target)
+            inst_targets = inst_targets.union(elm.inst_targets)
```

Comment:
```suggestion
            inst_targets |= elm.inst_targets
```
This is mutation and less expensive. This is not significant though.